### PR TITLE
fix: Use the right >/</>=/<= comparison operators for bitvectors in JS

### LINKED ISSUE #2672: JavaScript compiler: incorrect bignum comparison
Consider:

```
method Main() {
  print 24 as bv16 <= 1507 as bv16, "\n";
  assert 24 as bv16 <= 1507 as bv16;
}
```

This should print "true", which is what Go, Java, C#, and Python output. However, JavaScript prints "false".

### PR DESCRIPTION
**Pull request by cpitclaudel**
Closes #2672.

### CODE REVIEW

File: Test/git-issues/git-issue-2672.dfy
Comment (robin-aws) [inline Test/git-issues/git-issue-2672.dfy:7]: Can you use `%testDafnyForEachCompiler %s` instead? :)

File: Source/Dafny/Compilers/Compiler-js.cs
Comment (robin-aws) [inline Source/Dafny/Compilers/Compiler-js.cs:2015]: AFAICT we always `throw new cce.UnreachableException();` after such a contract assertion. That ensures control flow doesn't just continue in release builds.

**Review by robin-aws — CHANGES_REQUESTED** [review]

File: Test/git-issues/git-issue-2672.dfy
Comment (cpitclaudel) [reply Test/git-issues/git-issue-2672.dfy:7]: Can you remind me how to generate the output with that flag?

Comment (robin-aws) [reply Test/git-issues/git-issue-2672.dfy:7]: It should just be a single copy of the program's output, not including the "Dafny program verifier ..." line. So in this case just:

```
true true true true true true true true true
false false false false false false false false false
false false false false false false false false false
true true true true true true true true true
true true true true true true true true true
false false false false false false false false false
true true true
false false false
```

It occurs to me that having a flag on the `TestDafny` tool to generate the expect file would be super handy, let me know if I can add that. :)

Comment (cpitclaudel) [reply Test/git-issues/git-issue-2672.dfy:7]: Thanks!  Unfortunately it doesn't correctly detect unsupported features, so it fails on this example; I've reported the error at https://github.com/dafny-lang/dafny/issues/2718 , and I'm currently preparing a PR.  We can update this test file once that PR is merged, but I don't think we need to block on it.

**Review by robin-aws — APPROVED** [review]

**Review by robin-aws — APPROVED** [review]

**Pull request merged**
